docs/Modern-TypeEmission.md

```diff
@@ -0,0 +1,325 @@
+
+### Operation/Fragment selection-set data
+
+The shape of the data selected in a operation or fragment, following the [data-masking] rules. That is, excluding any data selected by fragment spreads, unless the `@relay(mask: false)` directive is used.
+
+In this example the emitted type-information describes the response data available to the operation’s render function.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Flow-->
+```javascript
+/**
+ * export type ExampleQueryResponse = {|
+ *   +artist: ?{|
+ *     +name: string
+ *   |}
+ * |}
+ */
+import type { ExampleQueryResponse } from "__generated__/ExampleQuery.graphql"
+
+<QueryRenderer
+  query={graphql`
+    query ExampleQuery {
+      artist(id: "banksy") {
+        name
+      }
+    }
+  `}
+  render={({ props }: { props?: ExampleQueryResponse }) => {
+    if (props) {
+      return props.artist && <div>{props.artist.name} is great!</div>
+    }
+    return <div>Loading</div>
+  }}
+/>
+```
+<!--TypeScript-->
+```javascript
+/**
+ * export type ExampleQueryResponse = {
+ *   readonly artist?: {
+ *     readonly name: string
+ *   }
+ * }
+ * export type ExampleQuery = {
+ *   readonly response: ExampleQueryResponse
+ * }
+ */
+import { ExampleQuery } from "__generated__/ExampleQuery.graphql"
+
+<QueryRenderer<ExampleQuery>
+  query={graphql`
+    query ExampleQuery {
+      artist(id: "banksy") {
+        name
+      }
+    }
+  `}
+  render={({ props }) => {
+    if (props) {
+      return props.artist && <div>{props.artist.name} is great!</div>
+    }
+    return <div>Loading</div>
+  }}
+/>
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+Similarly, in this example the emitted type-information describes the prop data that the container expects to receive.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Flow-->
+```javascript
+/**
+ * export type ExampleFragment_artist = {|
+ *   +name: string
+ * |}
+ */
+import type { ExampleFragment_artist } from "__generated__/ExampleFragment_artist.graphql"
+
+export const ExampleFragment = createFragmentContainer(
+  (props: { artist: ExampleFragment_artist }) => (
+    <div>About the artist: {props.artist.biography}</div>
+  ),
+  graphql`
+    fragment ExampleFragment_artist on Artist {
+      biography
+    }
+  `
+)
+```
+<!--TypeScript-->
+```javascript
+/**
+ * export type ExampleFragment_artist = {
+ *   readonly name: string
+ * }
+ */
+import { ExampleFragment_artist } from "__generated__/ExampleFragment_artist.graphql"
+
+export const ExampleFragment = createFragmentContainer(
+  (props: { artist: ExampleFragment_artist }) => (
+    <div>About the artist: {props.artist.biography}</div>
+  ),
+  graphql`
+    fragment ExampleFragment_artist on Artist {
+      biography
+    }
+  `
+)
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+### Fragment references
+
+The opaque identifier described in [data-masking] that a child container expects to receive from its parent, which represents the child container’s fragment spread inside the parent’s fragment.
```

Comment:
I guess, related to the above comment about incomplete sentences, is this a definition for fragment references? 

(Note: the technical term for incomplete sentences is "sentence fragment", so this is actually very on-brand 👍)